src/store/modules/ProfileModule.ts

```diff
@@ -87,4 +92,73 @@ export default {
             state.searchQuery = value.trim();
         },
     },
+
+    actions: <ActionTree<State, RootState>>{
+        async disableModsFromActiveProfile(
+            {dispatch},
+            params: {
+                mods: ManifestV2[],
+                onProgress?: (mod: ManifestV2) => void,
+            }
+        ) {
+            // TODO: manage active profile in Vuex.
+            const profile = Profile.getActiveProfile();
+
+            // Sanity check.
+            if (profile === undefined) {
+                throw new R2Error(
+                    'No active profile found',
+                    'Unable to disable mods when active profile is not set.'
+                )
+            }
+
+            await dispatch('disableModsFromProfile', {...params, profile});
+        },
+
+        async disableModsFromProfile(
+            {dispatch},
+            params: {
+                mods: ManifestV2[],
+                profile: Profile,
+                onProgress?: (mod: ManifestV2) => void,
+            }
+        ) {
+            const {mods, profile, onProgress} = params;
+            let lastSuccessfulUpdate: ManifestV2[] = [];
+
+            try {
+                // Disable mods on disk.
+                for (const mod of mods) {
+                    onProgress && onProgress(mod);
+
+                    if (!mod.isEnabled()) {
+                        continue;
+                    }
+
+                    const err = await ProfileInstallerProvider.instance.disableMod(mod, profile);
+                    if (err instanceof R2Error) {
+                        throw err;
+                    }
+                }
+
+                // Update mod list status to mods.yml and Vuex.
+                const updatedList = await ProfileModList.updateMods(mods, profile, (mod) => mod.disable());
+                if (updatedList instanceof R2Error) {
+                    throw updatedList;
+                } else {
+                    lastSuccessfulUpdate = updatedList;
+                }
+            } finally {
+                if (lastSuccessfulUpdate.length) {
```

Comment:
I guess this could technically be a bug: if for some reason the length becomes 0 during our mutations, we should still update the local mod list. I don't think it matters in the current codebase but commenting on it because I'd see it safer to check against null/undefined than 0 length in scenarios like this. A lack of a value isn't the same as an empty value after all.

Reply:
IIRC this was more or less copied from the Manager component. But I changed it `undefined` instead. This means that `ConflictManagementProvider.instance.resolveConflicts` never gets called if successful updates haven't been made. Hopefully that's the right thing to do.